### PR TITLE
add multi_npu_qwen3_dense tutorials

### PR DESCRIPTION
**Pull request by wind-all**
### What this PR does / why we need it?

This PR adds tutorials for the Qwen3-Dense series models, including the A2 and A3 series, and provides accuracy validation results.

### How was this patch tested?

- vLLM version: v0.12.0
- vLLM main: https://github.com/vllm-project/vllm/commit/ad32e3e19ccf0526cb6744a5fed09a138a5fb2f9

### CODE REVIEW

File: docs/source/tutorials/index.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ single_node_pd_disaggregation_llmdatadist
 multi_npu_qwen3_next
 multi_npu
 multi_npu_moge
+multi_npu_qwen3_dense
```
Comment (1092626063) [inline docs/source/tutorials/index.md]: The name needs to be uniformly changed to `Qwen3-Dense`

File: docs/source/tutorials/multi_npu_qwen3_dense.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
+# Multi-NPU (Qwen3-32B-W8A8)
```
Comment (1092626063) [inline docs/source/tutorials/multi_npu_qwen3_dense.md]: Modify the title to `Qwen3-Dense`, and you could take `Qwen3-32B` as an example

File: docs/source/tutorials/multi_npu_qwen3_dense.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+You can using our official docker image for supporting Qwen3 Dense models.
+Currently, we provide the all-in-one images `quay.io/ascend/vllm-ascend:v0.11.0rc2`、`quay.io/ascend/vllm-ascend:v0.11.0rc2-a3` and so on.[Download images](https://quay.io/repository/ascend/vllm-ascend?tab=tags)
+
+#### Docker Pull (by tag)
+
+`docker pull quay.io/ascend/vllm-ascend:v0.11.0rc2`
+
+`docker pull quay.io/ascend/vllm-ascend:v0.11.0rc2-a3`
+
+Start the docker image on your node, refer to [using docker](../installation.md#set-up-using-docker).
```
Comment (1092626063) [inline docs/source/tutorials/multi_npu_qwen3_dense.md]: please provide docker run command directly

File: docs/source/tutorials/index.md (hunk 1 above)
Comment (wind-all) [reply docs/source/tutorials/index.md]: Modified

File: docs/source/tutorials/multi_npu_qwen3_dense.md (hunk 2 above)
Comment (wind-all) [reply docs/source/tutorials/multi_npu_qwen3_dense.md]: Modified

File: docs/source/tutorials/multi_npu_qwen3_dense.md (hunk 3 above)
Comment (wind-all) [reply docs/source/tutorials/multi_npu_qwen3_dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+### Installation
+
+You can using our official docker image for supporting Qwen3 Dense models.
+Currently, we provide the all-in-one images `quay.io/ascend/vllm-ascend:v0.11.0rc2`、`quay.io/ascend/vllm-ascend:v0.11.0rc2-a3` and so on.[Download images](https://quay.io/repository/ascend/vllm-ascend?tab=tags)
+
+#### Docker Pull (by tag)
+
+`docker pull quay.io/ascend/vllm-ascend:v0.11.0rc2`
+
+`docker pull quay.io/ascend/vllm-ascend:v0.11.0rc2-a3`
```
Comment (1092626063) [inline docs/source/tutorials/Qwen3-Dense.md]: please do not hardcode vllm-ascend image version，it's better to use `|vllm_ascend_version|` bellow

Comment (wind-many) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+## Environment Preparation
+
+### Model Weight
+
+- `QWEN3-0.6B`(BF16 version): require 1 Atlas 800 A3 (64G × 16) node or 1 Atlas 800I A2 (64G × 8) node. [Download model weight](https://modelers.cn/models/Modelers_Park/Qwen3-0.6B)
+- `QWEN3-1.7B`(BF16 version): require 1 Atlas 800 A3 (64G × 16) node or 1 Atlas 800I A2 (64G × 8) node. [Download model weight](https://modelers.cn/models/Modelers_Park/Qwen3-1.7B)
+- `QWEN3-4B`(BF16 version): require 1 Atlas 800 A3 (64G × 16) node or 1 Atlas 800I A2 (64G × 8) node. [Download model weight](https://modelers.cn/models/Modelers_Park/Qwen3-4B)
+- `QWEN3-8B`(BF16 version): require 1 Atlas 800 A3 (64G × 16) node or 1 Atlas 800I A2 (64G × 8) node. [Download model weight](https://modelers.cn/models/Modelers_Park/Qwen3-8B)
+- `QWEN3-14B`(BF16 version): require 2 Atlas 800 A3 (64G × 16) node or 2 Atlas 800I A2 (64G × 8) node. [Download model weight](https://modelers.cn/models/Modelers_Park/Qwen3-14B)
+- `QWEN3-32B`(BF16 version): require 4 Atlas 800 A3 (64G × 16) nodes or 4 Atlas 800I A2 (64G × 8) nodes. [Download model weight](https://modelers.cn/models/Modelers_Park/Qwen3-32B)
```
Comment (menogrey) [inline docs/source/tutorials/Qwen3-Dense.md]: QWEN3->Qwen3, please modify it.

File: docs/source/tutorials/Qwen3-Dense.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+It is recommended to download the model weight to the shared directory of multiple nodes, such as `/root/.cache/`
+
+### Verify Multi-node Communication(Optional)
+
+If you want to deploy multi-node environment, you need to verify multi-node communication according to [verify multi-node communication environment](../installation.md#verify-multi-node-communication).
+
+### Installation
+
+You can using our official docker image for supporting Qwen3 Dense models.
+Currently, we provide the all-in-one images `quay.io/ascend/vllm-ascend:v0.11.0rc2`、`quay.io/ascend/vllm-ascend:v0.11.0rc2-a3` and so on.[Download images](https://quay.io/repository/ascend/vllm-ascend?tab=tags)
```
Comment (menogrey) [inline docs/source/tutorials/Qwen3-Dense.md]:
```suggestion
Currently, we provide the all-in-one images.[Download images](https://quay.io/repository/ascend/vllm-ascend?tab=tags)
```

File: docs/source/tutorials/Qwen3-Dense.md [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+
+#### **Attention**
+- /model/Qwen3-32B-W8A8 is the model path, replace this with your actual path.
+- v0.11.0rc2-a3 is image tag, replace this with your actual tag.
+- replace this with your actual port: '-p 8113:8113'.
+- replace this with your actual card: '--device /dev/davinci0'.
+
+```{code-block} bash
+# Update the vllm-ascend image
+export IMAGE=quay.io/ascend/vllm-ascend:v0.11.0rc2-a3
```
Comment (menogrey) [inline docs/source/tutorials/Qwen3-Dense.md]: You should modify it also.

File: docs/source/tutorials/Qwen3-Dense.md [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+-v /model/Qwen3-32B-W8A8:/model/Qwen3-32B-W8A8 \
+-p 8113:8113 \
+-it $IMAGE bash
+```
+
+### Setup environment variables
+
+```bash
+# Set vLLM to Engine V1
+export VLLM_USE_V1=1
```
Comment (menogrey) [inline docs/source/tutorials/Qwen3-Dense.md]: No need to `export VLLM_USE_V1=1` now. Delete it.

File: docs/source/tutorials/Qwen3-Dense.md [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+
+# set the NPU device number
+export ASCEND_RT_VISIBLE_DEVICES=0,1,2,3
+
+# Set the operator dispatch pipeline level to 1 and disable manual memory control in ACLGraph
+export TASK_QUEUE_ENABLE=1
+
+# Performance optimization of memory management
+# if os is Ubuntu
+apt update
```
Comment (menogrey) [inline docs/source/tutorials/Qwen3-Dense.md]: Too much installation guide for jemalloc2, refer to https://github.com/vllm-project/vllm-ascend/pull/4399/files

File: docs/source/tutorials/Qwen3-Dense.md (hunk 5 above)
Comment (wind-many) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md (hunk 6 above)
Comment (wind-many) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md (hunk 7 above)
Comment (wind-many) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md (hunk 8 above)
Comment (wind-many) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md (hunk 9 above)
Comment (wind-many) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

**Comment by menogrey** [comment]
LGTM, thanks for your contribution.

File: docs/source/tutorials/Qwen3-Dense.md [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+sed -i '/^name.*$/a sslverify=0' /etc/yum.repos.d/openEuler.repo
+yum install -y jemalloc
+# Add the LD_PRELOAD environment variable
+if [ -f /usr/lib/aarch64-linux-gnu/libjemalloc.so.2 ]; then
+    # On Ubuntu, first install with `apt install libjemalloc2`
+    export LD_PRELOAD=/usr/lib/aarch64-linux-gnu/libjemalloc.so.2:$LD_PRELOAD
+elif [ -f /usr/lib64/libjemalloc.so.2 ]; then
+    # On openEuler, first install with `yum install jemalloc`
+    export LD_PRELOAD=/usr/lib64/libjemalloc.so.2:$LD_PRELOAD
+fi
```
[the remark this answers is not in the document]
Comment (wind-all) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md [hunk 11]
```diff
@@ -0,0 +1,375 @@
...
+Qwen3 is the latest generation of large language models in Qwen series, offering a comprehensive suite of dense and mixture-of-experts (MoE) models. Built upon extensive training, Qwen3 delivers groundbreaking advancements in reasoning, instruction-following, agent capabilities, and multilingual support.
+
+Welcome to the tutorial on optimizing Qwen Dense models in the vLLM-Ascend environment. This guide will help you configure the most effective settings for your use case, with practical examples that highlight key optimization points. We will also explore how adjusting service parameters can maximize throughput performance across various scenarios.
+
+This document will show the main verification steps of the model, including supported features, feature configuration, environment preparation, accuracy and performance evaluation.
+
+The Qwen3 Dense models is first supported in [v0.8.4rc2](https://github.com/vllm-project/vllm-ascend/blob/main/docs/source/user_guide/release_notes.md#v084rc2---20250429)
+
+## **Node**
+This example requires version **v0.11.0rc2**.Earlier versions may lack certain features.
```
Comment (realliujiaxu) [inline docs/source/tutorials/Qwen3-Dense.md]: add space before "Earlier"

File: docs/source/tutorials/Qwen3-Dense.md [hunk 12]
```diff
@@ -0,0 +1,375 @@
...
+
+### 7. FullGraph Optimization
+ACLGraph offers several key optimizations to improve model execution efficiency. By replaying the entire model execution graph at once, we significantly reduce dispatch latency compared to multiple smaller replays. This approach also stabilizes multi-device performance, as capturing the model as a single static graph mitigates dispatch fluctuations across devices. Additionally, consolidating graph captures frees up streams, allowing for the capture of more graphs and optimizing resource usage, ultimately leading to improved system efficiency and reduced overhead.
+
+The configuration compilation_config = { "cudagraph_mode": "FULL_DECODE_ONLY"} is used when starting the service. This setup is necessary to enable the aclgraph's full decode-only mode.
+
+### 8. Asynchronous Scheduling
+Asynchronous scheduling is a technique used to optimize inference efficiency. It allows non-blocking task scheduling to improve concurrency and throughput, especially when processing large-scale models.
+
+This optimization is enabled by setting --async-scheduling
```
Comment (realliujiaxu) [inline docs/source/tutorials/Qwen3-Dense.md]: change --async-scheduling to `--async-scheduling`

File: docs/source/tutorials/Qwen3-Dense.md [hunk 13]
```diff
@@ -0,0 +1,375 @@
...
+
+### 1.Prefetch Buffer Size
+Setting the right prefetch buffer size is essential for optimizing weight loading and the size of this prefetch buffer is directly related to the time that can be hidden by vector computations. To achieve a near-perfect overlap between the prefetch and computation streams, you can flexibly adjust the buffer size by profiling and observing the degree of overlap at different buffer sizes.
+
+For example, in the real-world scenario mentioned above, I set the prefetch buffer size for the gate_up_proj and down_proj in the MLP to 18MB. The reason for this is that, at this value, the vector computations of RMSNorm and SiLU can effectively hide the prefetch stream, thereby accelerating the Matmul computations of the two linear layers.
+
+### 2.Max-model-len
```
Comment (realliujiaxu) [inline docs/source/tutorials/Qwen3-Dense.md]: `--max-model-len` is a parameter that depends on application requirements and model capabilities, not a  parameter that can be tuned to optimize performance. I suggest deleing this chapter.

File: docs/source/tutorials/Qwen3-Dense.md [hunk 14]
```diff
@@ -0,0 +1,375 @@
...
+
+If this list is not manually specified, it will be filled with a series of evenly distributed values, which typically ensures good performance. However, if you want to fine-tune it further, manually specifying the values will yield better results. This is because if the batch size falls between two sizes, the framework will automatically pad the token count to the larger size. This often leads to actual performance deviating from the expected or even degrading.
+
+Therefore, like the above real-world scenario, when adjusting the benchmark request concurrency, we always ensure that the concurrency is actually included in the cudagraph_capture_sizes list. This way, during the decode phase, padding operations are essentially avoided, ensuring the reliability of the experimental data.
+
+It’s important to note that if you enable FlashComm_v1, the values in this list must be integer multiples of the TP count. Any values that do not meet this condition will be automatically filtered out. Therefore, I recommend incrementally adding concurrency based on the TP count after enabling FlashComm_v1.
```
Comment (realliujiaxu) [inline docs/source/tutorials/Qwen3-Dense.md]: TP count -> TP size

File: docs/source/tutorials/Qwen3-Dense.md (hunk 11 above)
Comment (wind-all) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md (hunk 12 above)
Comment (wind-all) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md (hunk 13 above)
Comment (wind-all) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

File: docs/source/tutorials/Qwen3-Dense.md (hunk 14 above)
Comment (wind-all) [reply docs/source/tutorials/Qwen3-Dense.md]: Modified

**Comment by realliujiaxu** [comment]
LGTM

**Review by realliujiaxu — APPROVED** [review]

**Review by wind-many — COMMENTED** [review]
ok

**Review by wangxiyuan — APPROVED** [review]
